detection-rules/attachment_with_html_smuggling_fromcharcode_and_others.yml

```diff
@@ -0,0 +1,24 @@
+name: "Attachment: HTML smuggling with fromCharCode and other signals"
+description: |
+  Recursively scans files and archives to detect HTML smuggling techniques.
+references:
+  - "https://www.microsoft.com/security/blog/2021/11/11/html-smuggling-surges-highly-evasive-loader-technique-increasingly-used-in-banking-malware-targeted-attacks/"
+type: "rule"
+source: |
+  type.inbound
+  and any(attachments,
+      .file_extension in~ ("html","htm",
+                          "7z","bz2","gz","rar","tar","zip","zipx","iso","img")
+      and any(beta.binexplode(.),
+          any(.scan.javascript.identifiers, . == "location")
+          and any(.scan.javascript.identifiers, . == "charCodeAt")
+          and any(.scan.javascript.identifiers, . == "fromCharCode")
+          and any(.scan.javascript.identifiers, . == "indexOf")
+          and any(.scan.javascript.keywords, . == "try")
+          and any(.scan.javascript.keywords, . == "catch")
+          and length(.scan.javascript.identifiers) < 100  
```

Comment:
> I imagine that `location`, `indexOf`, `try`, `catch` are pretty noisy indicators. It's a little tricky because what's "normal" for an HTML attachment and what's normal HTML for the web are different.

the combination works with 100% TPs in testing so far

> Since these are all `and`ed together, I wonder if it's almost too specific and might miss other campaigns that have some but not all of these? I suspect that `try`/`catch` aren't totally necessary

i bet it is overly specific. i'm ok with that for the initial version of this rule though, since this gives us coverage on smuggling attacks that we're otherwise missing with 0 FPs. i think we could make improvements over time. re: try/catch, this is something i've seen quite often (and used to use in payloads I built), as a sort of catch-all to silently fail and hide any errors from the user

> Also, since these are all `==` checks, this syntax is an option: `and any(.scan.javascript.identifiers, . == "charCodeAt")` → `and "charCodeAt" in .scan.javascript.identifiers`

👍 can update to that. i originally wanted to use `all[]` but had hit that editor bug

Reply:
@rw-access updated!